### DISCUSSION TITLE
Accept string in the body

### DISCUSSION THREAD

**Discussion by roschel** [fastapi/fastapi#9010] [Questions] [answered]
### Description

Is there any way to input a string in the body without pydantic validation crashes?

![image](https://user-images.githubusercontent.com/52433168/108067761-0344f200-7040-11eb-8d48-9b5c5c5b51c6.png)

### Environment

* OS: Linux
* FastAPI Version 0.60.1
* Pydantic Version pydantic 1.6.1
* Starlette Version 0.13.6

**Comment by krishnardt** [comment]
As per my knowledge, it can be achieved by sending string  through request body.

**Comment by insomnes** [comment]
@roschel if you want accept just string, as @krishnardt mentioned, you can use `Body`:
```Python3
from fastapi import FastAPI, Body

app = FastAPI()


@app.post("/")
def main(body: str = Body(...)):
    return {"message": body}

```

![image](https://user-images.githubusercontent.com/16958893/108071660-0ad3bd80-7077-11eb-8162-8af3d9cea92b.png)

**Comment by ycd** [comment]
Are you trying to skip input validation? 

Is that what you mean by "without pydantic validation crashes" ?

**Comment by roschel** [comment]
Something like that. I need to read a json wrapped by quotes and Pydantic doesn't allow it.

The only way I found to do it was getting the request:
```python
async def post(request: Request):

    data = await request.body()
```

**Comment by ycd** [comment]
This confused me: _"need to read a json wrapped by quotes"_, can you share the request body?

The request body in the picture, seems pretty valid to me.

**Comment by roschel** [comment]
Request body in the picture, which returns the follow error:
```python
"
{
  "name": "xxxx",
  "code": "xxxx",
  "state": "xx",
  "city": "xxxxxxx",
  "active": true,
}
"
```

Error (status code 422 - Unprocessable Entity):
```
{
  "detail": [
    {
      "loc": [
        "body",
        1
      ],
      "msg": "Invalid control character at: line 1 column 2 (char 1)",
      "type": "value_error.jsondecode",
      "ctx": {
        "msg": "Invalid control character at",
        "doc": "\"\n{\n  \"name\": \"xxxx\",\n  \"code\": \"xxxx\",\n  \"state\": \"xx\",\n  \"city\": \"xxxxxxx\",\n  \"active\": true,\n}\n\"",
        "pos": 1,
        "lineno": 1,
        "colno": 2
      }
    }
  ]
}
```

**Comment by krishnardt** [comment]
@roschel can you send pydantic class too, that you are trying to validate the request it with?

**Comment by ycd** [comment]
I think you meant to use single quotes instead of double. This works fine with your example.

```curl
curl -X POST http://127.0.0.1:8000/dummy -d '{
  "name": "string",
  "code": "string",
  "state": "string",
  "city": "string",
  "active": true
}'
```

**Comment by roschel** [comment]
No, it has to be double quotes.
I've tried with single quote, but returns the same error.

My Pydantic class:
```python
class Example(BaseModel):
    name: str
    code: str
    state: str
    city: str = None
    active: bool = True
```

**Comment by krishnardt** [comment]
As far as I know,
It is not possible if the request validation doesnt match with the input data.
In your case, you want to pass entire data as string... create a pydantic model accordingly.

`class Example(BaseModel):
    data: str = None`

with the given pydantic class and with your given input in double quotes, it is not possible.

`{
  "data": "{'name': 'string','code': 'string','state': 'string','city': 'string','active': true}"
}`
You can try this one with the mentioned pydantic class.

**Comment by ycd** [comment]
> No, it has to be double quotes.

It does not, if you are going to use double quotes, you need to escape all the double quotes inside it. 

Are you using `-H "Content-Type: application/json"`?

**Comment by Kkkassini** [comment]
tested with double quotes in list of string, returned 422 error. This is quite stupid

**Comment by MSchwesig** [comment, marked as answer]
I think the problem lies elsewhere:
if the string contains a newline, the check fails.
The whole string must be in double quotes. All double quotes inside the string must be escaped with a` \ `
So a string like:
```
" 
Mello  Hello
"
```
will fail. but 
`"Mello Hello" `
is valid.

And for the escape part:
```
" My "Mello"  Hello"
```
is invalid.

```
" My \"Mello\"  Hello"
```
is valid.
